### ISSUE TITLE
Cross-targeting build /t:Clean;Build should work

### ISSUE THREAD

**Issue by nguerrera** [Microsoft/msbuild#1065]
See https://github.com/Microsoft/msbuild/pull/1063#issuecomment-248770988

**Comment by rainersigwald** [comment]
Another issue to consider here is build extensions that hook onto the `Build` target, that doesn't fire in the outer cross-targeting Rebuild. See https://github.com/NuGet/Home/issues/4475, and #1663.

**Comment by rainersigwald** [comment]
NuGet would appreciate a fix for this to avoid having to take a change for NuGet/Home#4475. Bringing it in for consideration (@rrelyea).

**Comment by rainersigwald** [comment]
Ok, so the core issue is that the MSBuild engine ensures that a given target is executed only once (per project per global properties). Since crosstargeting build and clean work by setting a property and then invoking the inner builds with `DispatchToInnerBuilds`, only the first (clean) inner build will execute.

For the specific case of `/t:clean;build`, we could just duplicate to have `DispatchToInnerBuildsForClean` and `DispatchToInnerBuilds`, so the former gets invoked on clean and the latter for build.

I think that's actually good enough here: a user could still specify a set of inner targets as a property on the command line, and they'd get passed through the normal `DispatchToInnerBuilds`. It'd be broken if they did `/p:InnerTargets=X /t:Clean;Build` but that's broken today too.

**Comment by nguerrera** [comment]
@rainersigwald Sounds good. Any chance you could make the change? If not, I don't think I can get to it until Monday.

**Comment by rainersigwald** [comment]
Spoke to @rrelyea offline. The NuGet scenario doesn't meet the bar, so we're pushing the MSBuild fix out of RTW.

**Comment by nguerrera** [comment]
Ok
